exercise/exercise_models.py

```diff
@@ -443,29 +443,37 @@ def one_has_access(self, students, when=None):
         in consideration.
         """
         timing,d = self.get_timing(students, when or timezone.now())
+
+        # Use format that shows the hours and minutes
+        # TODO: This should use formats set in specific language settings
+        if get_language() == 'fi':
+            time_format = 'j.n.Y l H:i'
+        else:
+            time_format = 'D, M j Y, P'
+        formatted_time = date_format(d, time_format)
```

Comment:
Setting the `time_format` manually is not the cleanest way to do this. The `DATETIME_FORMAT` set in the language files of the `aplus` folder should be used instead.